src/transformers/modeling_utils.py

```diff
@@ -128,7 +128,18 @@ def invert_attention_mask(self, encoder_attention_mask: Tensor) -> Tensor:
         # encoder_extended_attention_mask = (encoder_extended_attention_mask ==
         # encoder_extended_attention_mask.transpose(-1, -2))
         encoder_extended_attention_mask = encoder_extended_attention_mask.to(dtype=self.dtype)  # fp16 compatibility
-        encoder_extended_attention_mask = (1.0 - encoder_extended_attention_mask) * -1e9
+
+        if self.dtype == torch.float16:
```

Comment:
might be simpler to say `mask_value = 1e-9 if self.dtype == torch.float32 else 1e-4`.
People might use other dtypes down the road and it doesn't seem like this should be the function that prevents that.

Or you could explicitly check for overflow.

Reply:
Can't we use another source of info for the `dtype` than `self.dtype`? (e.g. `encoder_attention_mask.dtype` maybe?)

In general there's no single dtype for a module as the soon-to-be-merged #4300 shows